Use Vue CLI

zcuric opened the pull request:
### This PR
- [x] Introduces Vue CLI to Tailor

#### Motivation

Main motivation for this PR is to align Tailor (and in future all other EE apps) with official tools that are available for our technology set. Also big trigger for this was [this tweet pool on Poi twitter profile](https://twitter.com/poi__js/status/1173248429616578560), which obviously states that Vue community is gravitating to the official tools rather to third-party solutions. 

Although I think Poi handles some things better then Vue CLI and some things in Vue CLI are made rather sloppy, I also think that by using the official tools we guard ourselves from being attached to third-party which could end up unmaintained. 

#### @vladimyr checklist 🗒

- [x] #354 being merged first and required changes done here too
- [x] Collection of former `poi` - now `vue-cli` plugins to get extracted to separate repo in order to decrease maintenance burden. That means keeping both `poi` versions until we migrate all our apps and this newly created `vue-cli` variants.
- [x] All development functionalities (nodemon, debugging both inside and outside of vscode, webpack inspection scripts) need to be implemented using `vue-cli` too being fully transparent to end users
- [x] Build dependencies **only** get thoroughly inspected and cleaned as part of transition
- [x] c/p of build log showing built artefacts needs to be pasted here showing that there isn't significant difference in build output in terms of chunking, sizes and stuff

zcuric committed:
Use Vue CLI (wip)

zcuric committed:
Update plugins and config to support Vue CLI

zcuric commented:
Closing this until gets official approval for merging.
/cc @vladimyr @abasic @kronicker

kronicker commented:
I'm in favor of switching to Vue CLI so I'm reopening this.

vladimyr commented on babel.config.js:
```diff
@@ -0,0 +1,6 @@
+module.exports = {
+  presets: [
+    ['@vue/app']
+  ],
```
Keep things vertically tight:
```js
  presets: [['@vue/app']],
  sourceType: ...
```
:bowtie:

vladimyr commented on build/plugins/brand.js:
```diff
@@ -7,14 +7,9 @@ const map = require('lodash/map');
 const merge = require('lodash/merge');
 const path = require('path');
 const stripJsonComments = require('strip-json-comments');
+const logger = require('../../server/shared/logger');
```
Application logger has no place inside build plugin :no_entry: 
Use what `vue-cli` provides to you, if there is anything provided.

vladimyr commented on build/plugins/brand.js:
```diff
@@ -39,35 +34,62 @@ const getStyleConfig = () => ({
+  serve.fn = (...args) => {
+    addBranding(api, pluginOptions, args[0]);
+    return serveFn(...args);
+  };
```
Wrapping function with lambda is risky business do to potential breaking of original `this` binding :warning: Use regular functions with `apply/call` binding.

vladimyr commented on build/plugins/brand.js:
```diff
@@ -39,35 +34,62 @@ const getStyleConfig = () => ({
+
+  api.service.projectOptions = projectOptions;
+  console.log(projectOptions.pages.index);
+  console.log(constants);
```
Clean up those `console.log`s :bowtie:

vladimyr commented on build/plugins/brand.js:
```diff
@@ -39,35 +34,62 @@ const getStyleConfig = () => ({
+
+  api.chainWebpack(webpackConfig => {
+    webpackConfig.plugin('DefinePlugin')
+      .use(require('webpack').DefinePlugin, [{
```
Not that it matters much but please move `require`s to top or at least outside of active code path.

vladimyr commented on build/plugins/brand.js:
```diff
@@ -39,35 +34,62 @@ const getStyleConfig = () => ({
+  };
+};
+
+function addBranding(api, { brand }, { brandConfig }) {
```
This `brandConfig` is cli flag? :thinking:

vladimyr commented on build/plugins/brand.js:
```diff
@@ -39,35 +34,62 @@ const getStyleConfig = () => ({
+  console.log(projectOptions.pages.index);
+  console.log(constants);
+
+  api.chainWebpack(webpackConfig => {
```
Don't be _over_-descriptive. Both `vue-cli` docs and `webpack-chain` docs call it simply `config` exclusively. I understand that this is more descriptive but it also makes people familiar with convention stop and think is that really a chainable config or something else.

vladimyr commented on build/plugins/clean-out-dir.js:
```diff
@@ -1,14 +1,19 @@
+  build.fn = (...args) => {
+    args[0].clean = false;
+    api.chainWebpack(webpackConfig => {
+      if (process.env.NODE_ENV !== 'production' || args[0]['no-clean']) return;
```
Move this out of way by defining `isProduction` var outside.

vladimyr commented on build/plugins/clean-out-dir.js:
```diff
@@ -1,14 +1,19 @@
+        .plugin('clean-out-dir')
+        .use(CleanWebpackPlugin, [cleanOutDir]);
+    });
+    return buildFn(...args);
```
Suggestion:
```js
const { build } = api.service.commands;
const _build = build.fn;
build.fn = function (options = {}) {
  // ...
  return _build.apply(this, arguments);
}

vladimyr commented on build/plugins/html-version-spec.js:
```diff
@@ -3,27 +3,19 @@
-  const { output } = config;
-  output.html = output.html || {};
-  Object.assign(output.html, { meta });
+  config.pages.index = { ...config.pages.index, meta };
```
Did you just hardcode single page: `index`? :fox_face:

vladimyr commented on build/plugins/stats.js:
```diff
@@ -1,16 +1,7 @@
-    { default: false, type: [Boolean] }
-  );
-};
+const logger = require('../../server/shared/logger');
```
See my previous comment regarding loggers.

vladimyr commented on package.json:
```diff
@@ -173,13 +172,29 @@
+      "./build/plugins/copy.js",
+      "./build/plugins/clean-out-dir.js",
+      "./build/plugins/stats.js",
+      "./build/plugins/html-version-spec.js"
```
Omit extensions if possible.

vladimyr commented on package.json:
```diff
@@ -1,42 +1,38 @@
   "name": "tailor",
   "version": "3.2.2",
-  "codename": "silk",
+  "private": true,
```
Hmm, while I don't mind rearranging stuff inside `package.json` that has nothing to do with `vue-cli` introduction...

vladimyr commented on package.json:
```diff
@@ -1,42 +1,38 @@
-    "add:admin": "node -r ./server/script/preflight ./server/script/addAdmin.js",
-    "invite:admin": "node -r ./server/script/preflight ./server/script/inviteAdmin.js",
+    "debug:server": "node --inspect-brk -r ./server/script/preflight ./server",
+    "dev:client": "poi --serve",
```
Preserve old script names & functionalities plus get rid of `poi`...

vladimyr commented on public/index.html:
```diff
@@ -0,0 +1,12 @@
+<!DOCTYPE html>
```
Use https://github.com/jaketrent/html-webpack-template instead.

zcuric replied:
Done.

vladimyr commented on vue.config.js:
```diff
@@ -0,0 +1,87 @@
+    //   .tap(([options]) => {
+    //     options.terserOptions.keep_fnames = true;
+    //     return [options];
+    //   });
```
This can be removed.

vladimyr commented on vue.config.js:
```diff
@@ -0,0 +1,87 @@
+
+    config
+      .plugin('dotenv')
+      .use(require.resolve('dotenv-webpack'));
```
Get rid of this. It magically includes env variables in bundle. If it makes wrong assumption we might leak secrets.

vladimyr commented on vue.config.js:
```diff
@@ -0,0 +1,87 @@
+      ...STORAGE_PATH && { '^/repository': { target: serverUrl } }
+    }
+    // Override using: `npm run dev:server -- --port <number>`
+  },
```
Make sure that you:
1. do not drop type hints
2. copy everything from `poi.config.js`
3. do not alter stuff if no needed

:bowtie:

vladimyr reviewed (changes requested):
This is obviously wip but it is a good start :+1:

vladimyr commented:
> I'm in favor of switching to Vue CLI so I'm reopening this.

That being said in order to get me on board following things need to happen
1. #354 being merged first and required changes done here too
2. Collection of former `poi` - now `vue-cli` plugins to get extracted to separate repo in order to decrease maintenance burden. That means keeping both `poi` versions until we migrate all our apps and this newly created `vue-cli` variants.
3. All development functionalities (nodemon, debugging both inside and outside of vscode, webpack inspection scripts) need to be implemented using `vue-cli` too being fully transparent to end users
4. Build dependencies **only** get thoroughly inspected and cleaned as part of transition
5. c/p of build log showing built artefacts needs to be pasted here showing that there isn't significant difference in build output in terms of chunking, sizes and stuff

**PS** I should probably convert that to checklist :stuck_out_tongue_winking_eye:

zcuric committed:
Merge branch 'develop' into tooling/poi-to-vue-cli

zcuric committed:
Lint

zcuric committed:
Merge branch 'develop' into tooling/poi-to-vue-cli

zcuric committed:
Update plugins

zcuric committed:
Update package.json

zcuric committed:
Update babel config and import polyfills

zcuric committed:
Update vue.config.js and remove poi.config.js

zcuric committed:
Revert unintended changes in package.json

zcuric committed:
Update devServer in vue.config.js

zcuric committed:
Use HtmlWebpackPlugin

zcuric committed:
Remove unused package

vladimyr committed:
Merge branch 'develop' into tooling/poi-to-vue-cli

vladimyr committed:
Add type hint to vue-cli config

vladimyr committed:
Fix `webpack:inspect-config` npm script

vladimyr committed:
Implement sass-resources plugin

vladimyr committed:
Tweak page configuration

vladimyr committed:
Refactor clean-out-dir plugin

vladimyr committed:
Refactor sass-resources plugin

vladimyr committed:
Refactor brand plugin

vladimyr committed:
Refactor html-version plugin

vladimyr committed:
Sort imports :nail_care:

vladimyr reviewed (approved):
Looks good codewise 👍

vladimyr committed:
Going back to default html template

- removing `html-webpack-template` until
  https://github.com/jaketrent/html-webpack-template/issues/61
  gets resolved

zcuric committed:
Merge branch 'develop' into tooling/poi-to-vue-cli

zcuric committed:
Update Vue CLI to latest version

zcuric committed:
Update aliases

zcuric commented:
Build log from Vue CLI

```shell
➜ npm run build     

> tailor@3.2.2 build /Users/zdravko/Code/tailor
> del-cli node_modules/.cache/vue-loader && vue-cli-service build

 INFO  tailor:brand-plugin  Using default brand configuration

 WARN  A new version of sass-loader is available. Please upgrade for best experience.
⠦  Building for production...

 WARNING  Compiled with 2 warnings                                                      9:24:55 AM

 warning  

asset size limit: The following asset(s) exceed the recommended size limit (244 KiB).
This can impact web performance.
Assets: 
  fonts/materialdesignicons-webfont.27cb2cf1.woff2 (256 KiB)
  fonts/materialdesignicons-webfont.043774d1.woff (365 KiB)
  fonts/materialdesignicons-webfont.9bfeb985.ttf (754 KiB)
  fonts/materialdesignicons-webfont.e971abae.eot (755 KiB)
  js/chunk-80a14528.2be14257.js (339 KiB)
  css/chunk-vendors.1a8519f8.css (292 KiB)
  js/chunk-vendors.6a66df8d.js (2.02 MiB)
  css/index.5d27a97f.css (398 KiB)
  js/index.25fad976.js (288 KiB)

 warning  

entrypoint size limit: The following entrypoint(s) combined asset size exceeds the recommended lim
it (244 KiB). This can impact web performance.
Entrypoints:
  index (2.98 MiB)
      css/chunk-vendors.1a8519f8.css
      js/chunk-vendors.6a66df8d.js
      css/index.5d27a97f.css
      js/index.25fad976.js


  File                                    Size              Gzipped

  dist/js/chunk-vendors.6a66df8d.js       2073.25 KiB       588.48 KiB
  dist/js/chunk-80a14528.2be14257.js      338.99 KiB        90.75 KiB
  dist/js/index.25fad976.js               287.65 KiB        67.23 KiB
  dist/js/chunk-4f1b2290.f18efd9f.js      116.66 KiB        32.32 KiB
  dist/js/chunk-2995abc9.efb86c0e.js      45.79 KiB         13.66 KiB
  dist/js/chunk-343b7010.7271df42.js      40.32 KiB         12.50 KiB
  dist/js/chunk-14058beb.59b93430.js      16.01 KiB         5.73 KiB
  dist/js/chunk-be1914cc.36b58350.js      9.58 KiB          3.46 KiB
  dist/js/chunk-40200d95.38765a05.js      8.43 KiB          2.76 KiB
  dist/js/chunk-2f4656cc.247134ab.js      8.40 KiB          2.60 KiB
  dist/js/chunk-7e977163.76fd1c63.js      7.89 KiB          2.96 KiB
  dist/js/chunk-391cfe32.b965e7fc.js      7.68 KiB          2.81 KiB
  dist/js/chunk-05f0119f.558407fc.js      7.64 KiB          2.47 KiB
  dist/js/chunk-68bf95b5.9f79ce9b.js      7.54 KiB          2.61 KiB
  dist/js/chunk-198fbb83.d9c24801.js      7.42 KiB          2.55 KiB
  dist/js/chunk-2bdea133.312d1e84.js      6.84 KiB          1.91 KiB
  dist/js/chunk-9a44f47e.db3134ed.js      6.79 KiB          2.62 KiB
  dist/js/chunk-9effe5a4.a9a80f65.js      6.57 KiB          2.06 KiB
  dist/js/chunk-349fc5e9.fa5a0925.js      6.49 KiB          2.30 KiB
  dist/js/chunk-d8d8fad4.1f7046f2.js      6.31 KiB          1.72 KiB
  dist/js/chunk-46e83674.f2b6d7e1.js      6.28 KiB          2.18 KiB
  dist/js/chunk-478b3b95.13f2aa59.js      6.22 KiB          2.20 KiB
  dist/js/chunk-3a7dd888.7b44423c.js      5.82 KiB          2.07 KiB
  dist/js/chunk-851c1fba.86508a94.js      5.50 KiB          1.93 KiB
  dist/js/chunk-53b000c4.9cc1f6fd.js      5.34 KiB          1.73 KiB
  dist/js/chunk-ab0c17fc.1b3bafc8.js      5.16 KiB          1.89 KiB
  dist/js/chunk-5b0d5e86.d60e63aa.js      4.62 KiB          1.51 KiB
  dist/js/chunk-184d0f37.7b1814af.js      4.58 KiB          1.95 KiB
  dist/js/chunk-ce78375e.9b777a1b.js      4.53 KiB          1.79 KiB
  dist/js/chunk-6d598a9f.7bb4c42b.js      4.13 KiB          1.57 KiB
  dist/js/chunk-0c3f9347.e58b3707.js      3.94 KiB          0.87 KiB
  dist/js/chunk-fa144f42.06edea0c.js      3.93 KiB          1.68 KiB
  dist/js/chunk-2797a37e.320dd4d7.js      3.88 KiB          1.61 KiB
  dist/js/chunk-744a8fd0.b02b7d17.js      3.85 KiB          1.57 KiB
  dist/js/chunk-392a0f68.5a785b73.js      3.66 KiB          1.40 KiB
  dist/js/chunk-128fa85a.267c512a.js      3.65 KiB          1.43 KiB
  dist/js/chunk-d929a35c.b31e86d1.js      3.62 KiB          1.49 KiB
  dist/js/chunk-7d071210.3c15f8f7.js      3.62 KiB          1.50 KiB
  dist/js/chunk-5e38ca20.f440b73d.js      3.46 KiB          1.40 KiB
  dist/js/chunk-18c17cb5.b4421d98.js      3.38 KiB          1.38 KiB
  dist/js/chunk-21577292.fa0473ab.js      3.25 KiB          1.41 KiB
  dist/js/chunk-6e5d958d.dedef4d2.js      3.23 KiB          1.38 KiB
  dist/js/chunk-9b44f082.0b43401b.js      3.23 KiB          1.42 KiB
  dist/js/chunk-00544e8c.8c8fb35b.js      2.84 KiB          1.19 KiB
  dist/js/chunk-1e260ea6.dae31005.js      2.81 KiB          1.23 KiB
  dist/js/chunk-7dc79851.3c0ac2ee.js      2.80 KiB          1.08 KiB
  dist/js/chunk-0c4e18da.e7fe331c.js      2.75 KiB          1.25 KiB
  dist/js/chunk-2037b004.3ad45234.js      2.70 KiB          0.99 KiB
  dist/js/chunk-f78bd36a.628e0029.js      2.69 KiB          1.05 KiB
  dist/js/chunk-2d210fde.8fbbb95c.js      2.33 KiB          1.04 KiB
  dist/js/chunk-9c09781c.6f4eb2cb.js      2.22 KiB          1.03 KiB
  dist/js/chunk-aca92f48.4bfdcf9e.js      2.15 KiB          1.04 KiB
  dist/js/chunk-d14575c2.d74a63fc.js      2.10 KiB          0.92 KiB
  dist/js/chunk-6c5a1a14.88d02364.js      2.03 KiB          0.90 KiB
  dist/js/chunk-3bf5ffd6.9c03bc28.js      2.03 KiB          0.91 KiB
  dist/js/chunk-38c51faa.02e2023b.js      1.89 KiB          0.92 KiB
  dist/js/chunk-668a13e6.60f5cd6c.js      1.81 KiB          0.78 KiB
  dist/js/chunk-70d04da0.eead23e1.js      1.74 KiB          0.77 KiB
  dist/js/chunk-24e83a30.c0a5ad3c.js      1.69 KiB          0.87 KiB
  dist/js/chunk-4abb8988.eaaf75d1.js      1.61 KiB          0.85 KiB
  dist/js/chunk-2d209408.d3e7f59c.js      1.55 KiB          0.81 KiB
  dist/js/chunk-d2e02600.cace0028.js      1.49 KiB          0.77 KiB
  dist/js/chunk-3d1b3011.9652afc5.js      1.48 KiB          0.79 KiB
  dist/js/chunk-96fa80ee.813f8627.js      1.46 KiB          0.78 KiB
  dist/js/chunk-2d0c4ab4.31a14ff4.js      1.44 KiB          0.68 KiB
  dist/js/chunk-3d33437a.5b699c2a.js      1.42 KiB          0.69 KiB
  dist/js/chunk-db0c7092.b30fbb0a.js      1.38 KiB          0.72 KiB
  dist/js/chunk-7ac5211a.628f1f2f.js      1.33 KiB          0.73 KiB
  dist/js/chunk-3afa9eae.e65d578b.js      0.99 KiB          0.51 KiB
  dist/js/chunk-2d0b728b.07e918d0.js      0.76 KiB          0.39 KiB
  dist/js/chunk-5d840c3f.edc1718f.js      0.75 KiB          0.47 KiB
  dist/js/chunk-6edce3bc.e075916b.js      0.72 KiB          0.45 KiB
  dist/js/chunk-0bfab3bc.775edc5d.js      0.71 KiB          0.44 KiB
  dist/js/chunk-258df7b4.34a545d4.js      0.49 KiB          0.33 KiB
  dist/js/chunk-2d0d0214.413bd555.js      0.19 KiB          0.17 KiB
  dist/js/chunk-7467d93b.922f1237.js      0.15 KiB          0.14 KiB
  dist/js/chunk-772249b9.ad941284.js      0.15 KiB          0.14 KiB
  dist/css/index.5d27a97f.css             398.00 KiB        64.82 KiB
  dist/css/chunk-vendors.1a8519f8.css     291.75 KiB        38.07 KiB
  dist/css/chunk-4f1b2290.548137f2.css    24.48 KiB         4.35 KiB
  dist/css/chunk-343b7010.902d346b.css    3.65 KiB          1.21 KiB
  dist/css/chunk-198fbb83.5967d5e8.css    2.85 KiB          0.80 KiB
  dist/css/chunk-d8d8fad4.10ab1d90.css    2.76 KiB          0.79 KiB
  dist/css/chunk-392a0f68.d6cce490.css    2.39 KiB          0.63 KiB
  dist/css/chunk-0c3f9347.a01a04d4.css    2.38 KiB          0.66 KiB
  dist/css/chunk-46e83674.d25b9009.css    2.23 KiB          0.70 KiB
  dist/css/chunk-349fc5e9.437bb43b.css    2.08 KiB          0.65 KiB
  dist/css/chunk-40200d95.7fcff510.css    2.04 KiB          0.61 KiB
  dist/css/chunk-05f0119f.7fcff510.css    2.04 KiB          0.61 KiB
  dist/css/chunk-5b0d5e86.60d8e42e.css    1.79 KiB          0.57 KiB
  dist/css/chunk-ab0c17fc.388b48b8.css    1.75 KiB          0.58 KiB
  dist/css/chunk-851c1fba.0b59f4a7.css    1.73 KiB          0.53 KiB
  dist/css/chunk-478b3b95.0b59f4a7.css    1.73 KiB          0.53 KiB
  dist/css/chunk-5e38ca20.de9428d3.css    1.64 KiB          0.54 KiB
  dist/css/chunk-68bf95b5.a7281d48.css    1.50 KiB          0.58 KiB
  dist/css/chunk-be1914cc.2435eabf.css    1.47 KiB          0.57 KiB
  dist/css/chunk-14058beb.b2b0a652.css    1.37 KiB          0.51 KiB
  dist/css/chunk-7e977163.0f213bfa.css    1.31 KiB          0.50 KiB
  dist/css/chunk-2f4656cc.a75b7052.css    1.30 KiB          0.47 KiB
  dist/css/chunk-6e5d958d.a8f793ac.css    1.23 KiB          0.48 KiB
  dist/css/chunk-00544e8c.a8f793ac.css    1.23 KiB          0.48 KiB
  dist/css/chunk-9a44f47e.673503ce.css    1.15 KiB          0.47 KiB
  dist/css/chunk-3a7dd888.b8ee4182.css    1.13 KiB          0.41 KiB
  dist/css/chunk-ce78375e.2feedae0.css    1.10 KiB          0.45 KiB
  dist/css/chunk-0bfab3bc.ad065620.css    1.09 KiB          0.29 KiB
  dist/css/chunk-6c5a1a14.56cc41a2.css    1.04 KiB          0.35 KiB
  dist/css/chunk-6d598a9f.847fefbf.css    1.03 KiB          0.43 KiB
  dist/css/chunk-9b44f082.ee65ce81.css    1.02 KiB          0.42 KiB
  dist/css/chunk-1e260ea6.ee65ce81.css    1.02 KiB          0.42 KiB
  dist/css/chunk-80a14528.d7cb2d37.css    1.02 KiB          0.36 KiB
  dist/css/chunk-2037b004.1762f5eb.css    0.98 KiB          0.31 KiB
  dist/css/chunk-3d33437a.76c5965b.css    0.95 KiB          0.35 KiB
  dist/css/chunk-391cfe32.27fe6b23.css    0.84 KiB          0.34 KiB
  dist/css/chunk-3bf5ffd6.45acc45d.css    0.82 KiB          0.32 KiB
  dist/css/chunk-70d04da0.41c8d2c9.css    0.81 KiB          0.36 KiB
  dist/css/chunk-2995abc9.1c6ebefe.css    0.74 KiB          0.33 KiB
  dist/css/chunk-7ac5211a.73ec5b6b.css    0.74 KiB          0.33 KiB
  dist/css/chunk-4abb8988.73ec5b6b.css    0.74 KiB          0.33 KiB
  dist/css/chunk-184d0f37.ea2ed37c.css    0.73 KiB          0.32 KiB
  dist/css/chunk-fa144f42.ea2ed37c.css    0.73 KiB          0.32 KiB
  dist/css/chunk-18c17cb5.0f649ab0.css    0.62 KiB          0.30 KiB
  dist/css/chunk-744a8fd0.0f649ab0.css    0.62 KiB          0.30 KiB
  dist/css/chunk-668a13e6.bc163647.css    0.58 KiB          0.30 KiB
  dist/css/chunk-d14575c2.bc163647.css    0.58 KiB          0.30 KiB
  dist/css/chunk-21577292.c1dcb9f8.css    0.54 KiB          0.29 KiB
  dist/css/chunk-5d840c3f.b0ddf25e.css    0.47 KiB          0.24 KiB
  dist/css/chunk-7d071210.286b1810.css    0.46 KiB          0.27 KiB
  dist/css/chunk-3d1b3011.014022c5.css    0.46 KiB          0.26 KiB
  dist/css/chunk-d929a35c.971b94e8.css    0.45 KiB          0.26 KiB
  dist/css/chunk-2797a37e.971b94e8.css    0.45 KiB          0.26 KiB
  dist/css/chunk-7dc79851.abe3948c.css    0.40 KiB          0.24 KiB
  dist/css/chunk-0c4e18da.2c14ec6b.css    0.40 KiB          0.22 KiB
  dist/css/chunk-7467d93b.2c14ec6b.css    0.40 KiB          0.22 KiB
  dist/css/chunk-9c09781c.fcdfd413.css    0.38 KiB          0.22 KiB
  dist/css/chunk-9effe5a4.d9e3ef03.css    0.36 KiB          0.20 KiB
  dist/css/chunk-aca92f48.b4f629d7.css    0.35 KiB          0.21 KiB
  dist/css/chunk-38c51faa.b4f629d7.css    0.35 KiB          0.21 KiB
  dist/css/chunk-772249b9.3c3b16e6.css    0.34 KiB          0.19 KiB
  dist/css/chunk-53b000c4.3c3b16e6.css    0.34 KiB          0.19 KiB
  dist/css/chunk-2bdea133.3c3b16e6.css    0.34 KiB          0.19 KiB
  dist/css/chunk-128fa85a.570f2518.css    0.31 KiB          0.18 KiB
  dist/css/chunk-d2e02600.c7c5ccad.css    0.24 KiB          0.16 KiB
  dist/css/chunk-24e83a30.a67d8458.css    0.22 KiB          0.17 KiB
  dist/css/chunk-db0c7092.a67d8458.css    0.22 KiB          0.17 KiB
  dist/css/chunk-96fa80ee.01535477.css    0.08 KiB          0.10 KiB
  dist/css/chunk-6edce3bc.98a7b6f9.css    0.07 KiB          0.09 KiB
  dist/css/chunk-258df7b4.98a7b6f9.css    0.07 KiB          0.09 KiB

  Images and other types of assets omitted.

 DONE  Build complete. The dist directory is ready to be deployed.
 INFO  Check out deployment instructions at https://cli.vuejs.org/guide/deployment.html
```

zcuric commented:
### Build log from poi
```
✔ Build completed in 50.7s

file                                                    size     gzipped
dist/assets/css/0.33e455dc.chunk.css                    632 B    301 B
dist/assets/css/1.921b75a5.chunk.css                    3.74 kB  1.24 kB
dist/assets/css/10.2f5c10c9.chunk.css                   2.19 kB  703 B
dist/assets/css/11.40a5b3d1.chunk.css                   1.5 kB   585 B
dist/assets/css/12.476e3020.chunk.css                   2.02 kB  652 B
dist/assets/css/13.59a6a59c.chunk.css                   1.84 kB  587 B
dist/assets/css/14.069d96d7.chunk.css                   2.68 kB  777 B
dist/assets/css/15.2711dafb.chunk.css                   347 B    234 B
dist/assets/css/16.301a81b6.chunk.css                   1.56 kB  511 B
dist/assets/css/17.d5139697.chunk.css                   1.63 kB  520 B
dist/assets/css/18.19880a09.chunk.css                   1.16 kB  424 B
dist/assets/css/19.1bd7fa3a.chunk.css                   1.86 kB  611 B
dist/assets/css/2.9e962089.chunk.css                    1.36 kB  512 B
dist/assets/css/20.9a7102fc.chunk.css                   368 B    203 B
dist/assets/css/21.5e784d00.chunk.css                   1.71 kB  574 B
dist/assets/css/22.b1f724db.chunk.css                   1.09 kB  455 B
dist/assets/css/23.c87269dd.chunk.css                   956 B    428 B
dist/assets/css/24.04933eef.chunk.css                   1.56 kB  511 B
dist/assets/css/25.8d733627.chunk.css                   394 B    229 B
dist/assets/css/26.b1754243.chunk.css                   635 B    305 B
dist/assets/css/27.863a3beb.chunk.css                   358 B    215 B
dist/assets/css/28.78179b02.chunk.css                   748 B    325 B
dist/assets/css/29.0ceb53f2.chunk.css                   1 kB     413 B
dist/assets/css/3.0bb6c55d.chunk.css                    24.9 kB  4.42 kB
dist/assets/css/30.cc20fa4b.chunk.css                   635 B    305 B
dist/assets/css/31.25d2f1c7.chunk.css                   76 B     93 B
dist/assets/css/32.d3ce1059.chunk.css                   1.24 kB  476 B
dist/assets/css/33.c642d77f.chunk.css                   229 B    169 B
dist/assets/css/34.846dd07f.chunk.css                   592 B    304 B
dist/assets/css/35.1a26aa46.chunk.css                   622 B    307 B
dist/assets/css/36.58642937.chunk.css                   748 B    325 B
dist/assets/css/37.d82558fb.chunk.css                   313 B    180 B
dist/assets/css/38.bfab4966.chunk.css                   1 kB     413 B
dist/assets/css/39.d96d1c4c.chunk.css                   76 B     93 B
dist/assets/css/4.a0c79505.chunk.css                    1 kB     355 B
dist/assets/css/40.d46b1a89.chunk.css                   1.24 kB  476 B
dist/assets/css/41.5a96a146.chunk.css                   229 B    169 B
dist/assets/css/42.4600ca90.chunk.css                   592 B    304 B
dist/assets/css/44.e2ff0a22.chunk.css                   1.17 kB  498 B
dist/assets/css/45.3325579d.chunk.css                   864 B    343 B
dist/assets/css/46.2eb15af1.chunk.css                   1 kB     463 B
dist/assets/css/47.5e259f41.chunk.css                   357 B    242 B
dist/assets/css/48.e116c6ea.chunk.css                   347 B    234 B
dist/assets/css/49.62f2c2db.chunk.css                   323 B    180 B
dist/assets/css/5.4f848018.chunk.css                    2.09 kB  788 B
dist/assets/css/50.0bdb3270.chunk.css                   85 B     101 B
dist/assets/css/51.78404f24.chunk.css                   1.59 kB  537 B
dist/assets/css/52.af648822.chunk.css                   485 B    248 B
dist/assets/css/53.663b002e.chunk.css                   358 B    215 B
dist/assets/css/54.7d16aa54.chunk.css                   518 B    288 B
dist/assets/css/55.4a13d82d.chunk.css                   412 B    246 B
dist/assets/css/56.3b931e6a.chunk.css                   410 B    255 B
dist/assets/css/57.cb377600.chunk.css                   1.07 kB  354 B
dist/assets/css/58.159de267.chunk.css                   1 kB     311 B
dist/assets/css/59.267b619c.chunk.css                   2.29 kB  643 B
dist/assets/css/6.802e4e0a.chunk.css                    2.22 kB  746 B
dist/assets/css/60.f1663797.chunk.css                   844 B    326 B
dist/assets/css/61.cae06eca.chunk.css                   757 B    333 B
dist/assets/css/62.3d1f698f.chunk.css                   611 B    270 B
dist/assets/css/63.545663eb.chunk.css                   244 B    167 B
dist/assets/css/64.a9fbf8e4.chunk.css                   968 B    354 B
dist/assets/css/65.545c8ead.chunk.css                   323 B    180 B
dist/assets/css/66.e2c68242.chunk.css                   309 B    193 B
dist/assets/css/7.9ba7f3af.chunk.css                    1.33 kB  565 B
dist/assets/css/76.b3f8a942.chunk.css                   309 B    193 B
dist/assets/css/77.62926c8d.chunk.css                   323 B    180 B
dist/assets/css/8.c7f21858.chunk.css                    1.84 kB  587 B
dist/assets/css/9.810f5329.chunk.css                    1.34 kB  486 B
dist/assets/css/app.da2a5545.chunk.css                  402 kB   64.8 kB
dist/assets/css/chunk-vendors.ff8c2b62.chunk.css        257 kB   36.3 kB
dist/assets/fonts/materialdesignicons-webfont.eot       773 kB   374 kB
dist/assets/fonts/materialdesignicons-webfont.ttf       772 kB   374 kB
dist/assets/fonts/materialdesignicons-webfont.woff      373 kB   364 kB
dist/assets/fonts/materialdesignicons-webfont.woff2     262 kB   261 kB
dist/assets/img/default-favicon.ico                     5.43 kB  1.34 kB
dist/assets/img/default-logo-compact.svg                820 B    454 B
dist/assets/img/default-logo-full.svg                   820 B    454 B
dist/assets/js/0.33e455dc.chunk.js                      43.4 kB  12.6 kB
dist/assets/js/0.33e455dc.chunk.js.LICENSE              2.52 kB  772 B
dist/assets/js/1.921b75a5.chunk.js                      40.9 kB  12.5 kB
dist/assets/js/1.921b75a5.chunk.js.LICENSE              187 B    169 B
dist/assets/js/10.2f5c10c9.chunk.js                     6.21 kB  2.11 kB
dist/assets/js/11.40a5b3d1.chunk.js                     7.57 kB  2.57 kB
dist/assets/js/12.476e3020.chunk.js                     6.46 kB  2.19 kB
dist/assets/js/13.59a6a59c.chunk.js                     7.72 kB  2.37 kB
dist/assets/js/14.069d96d7.chunk.js                     6.55 kB  1.64 kB
dist/assets/js/15.2711dafb.chunk.js                     3.85 kB  1.54 kB
dist/assets/js/16.301a81b6.chunk.js                     6.25 kB  2.12 kB
dist/assets/js/17.d5139697.chunk.js                     4.65 kB  1.45 kB
dist/assets/js/18.19880a09.chunk.js                     5.78 kB  2.02 kB
dist/assets/js/19.1bd7fa3a.chunk.js                     3.65 kB  1.34 kB
dist/assets/js/2.9e962089.chunk.js                      16.1 kB  5.61 kB
dist/assets/js/20.9a7102fc.chunk.js                     6.46 kB  1.97 kB
dist/assets/js/21.5e784d00.chunk.js                     5.16 kB  1.82 kB
dist/assets/js/22.b1f724db.chunk.js                     4.54 kB  1.76 kB
dist/assets/js/23.c87269dd.chunk.js                     4.05 kB  1.47 kB
dist/assets/js/24.04933eef.chunk.js                     5.58 kB  1.85 kB
dist/assets/js/25.8d733627.chunk.js                     2.13 kB  968 B
dist/assets/js/26.b1754243.chunk.js                     3.81 kB  1.48 kB
dist/assets/js/27.863a3beb.chunk.js                     2.09 kB  976 B
dist/assets/js/28.78179b02.chunk.js                     4.52 kB  1.86 kB
dist/assets/js/29.0ceb53f2.chunk.js                     3.19 kB  1.35 kB
dist/assets/js/3.0bb6c55d.chunk.js                      122 kB   33.8 kB
dist/assets/js/30.cc20fa4b.chunk.js                     3.38 kB  1.31 kB
dist/assets/js/31.25d2f1c7.chunk.js                     599 B    377 B
dist/assets/js/32.d3ce1059.chunk.js                     3.19 kB  1.31 kB
dist/assets/js/33.c642d77f.chunk.js                     1.6 kB   804 B
dist/assets/js/34.846dd07f.chunk.js                     2.05 kB  858 B
dist/assets/js/35.1a26aa46.chunk.js                     1.71 kB  716 B
dist/assets/js/36.58642937.chunk.js                     3.92 kB  1.6 kB
dist/assets/js/37.d82558fb.chunk.js                     3.59 kB  1.4 kB
dist/assets/js/38.bfab4966.chunk.js                     2.81 kB  1.17 kB
dist/assets/js/39.d96d1c4c.chunk.js                     387 B    270 B
dist/assets/js/4.a0c79505.chunk.js                      322 kB   90.1 kB
dist/assets/js/4.a0c79505.chunk.js.LICENSE              12.6 kB  1.42 kB
dist/assets/js/40.d46b1a89.chunk.js                     2.83 kB  1.13 kB
dist/assets/js/41.5a96a146.chunk.js                     1.31 kB  668 B
dist/assets/js/42.4600ca90.chunk.js                     1.79 kB  735 B
dist/assets/js/43.d9863ace.chunk.js                     906 B    459 B
dist/assets/js/44.e2ff0a22.chunk.js                     7.85 kB  2.85 kB
dist/assets/js/45.3325579d.chunk.js                     7.58 kB  2.69 kB
dist/assets/js/46.2eb15af1.chunk.js                     6.75 kB  2.54 kB
dist/assets/js/47.5e259f41.chunk.js                     3.61 kB  1.44 kB
dist/assets/js/48.e116c6ea.chunk.js                     3.62 kB  1.45 kB
dist/assets/js/49.62f2c2db.chunk.js                     6.51 kB  1.79 kB
dist/assets/js/5.4f848018.chunk.js                      11 kB    3.75 kB
dist/assets/js/50.0bdb3270.chunk.js                     1.36 kB  701 B
dist/assets/js/51.78404f24.chunk.js                     3.49 kB  1.34 kB
dist/assets/js/52.af648822.chunk.js                     678 B    413 B
dist/assets/js/53.663b002e.chunk.js                     1.85 kB  868 B
dist/assets/js/54.7d16aa54.chunk.js                     3.25 kB  1.37 kB
dist/assets/js/55.4a13d82d.chunk.js                     2.81 kB  1.03 kB
dist/assets/js/56.3b931e6a.chunk.js                     1.42 kB  719 B
dist/assets/js/57.cb377600.chunk.js                     2.07 kB  876 B
dist/assets/js/58.159de267.chunk.js                     2.75 kB  962 B
dist/assets/js/59.267b619c.chunk.js                     4.24 kB  840 B
dist/assets/js/6.802e4e0a.chunk.js                      7.43 kB  2.47 kB
dist/assets/js/60.f1663797.chunk.js                     2.01 kB  864 B
dist/assets/js/61.cae06eca.chunk.js                     1.27 kB  663 B
dist/assets/js/62.3d1f698f.chunk.js                     619 B    375 B
dist/assets/js/63.545663eb.chunk.js                     1.42 kB  689 B
dist/assets/js/64.a9fbf8e4.chunk.js                     1.38 kB  656 B
dist/assets/js/65.545c8ead.chunk.js                     5.13 kB  1.64 kB
dist/assets/js/66.e2c68242.chunk.js                     2.67 kB  1.18 kB
dist/assets/js/67.9f962506.chunk.js                     3.04 kB  1.11 kB
dist/assets/js/68.3bab63a4.chunk.js                     1.47 kB  745 B
dist/assets/js/69.69514f3d.chunk.js                     118 B    120 B
dist/assets/js/7.9ba7f3af.chunk.js                      9.56 kB  3.34 kB
dist/assets/js/70.627f15e4.chunk.js                     1.4 kB   646 B
dist/assets/js/71.e51dbb88.chunk.js                     2.26 kB  995 B
dist/assets/js/72.71311fb0.chunk.js                     699 B    339 B
dist/assets/js/76.b3f8a942.chunk.js                     84 B     87 B
dist/assets/js/77.62926c8d.chunk.js                     84 B     87 B
dist/assets/js/8.c7f21858.chunk.js                      8.46 kB  2.65 kB
dist/assets/js/9.810f5329.chunk.js                      8.26 kB  2.48 kB
dist/assets/js/app.da2a5545.chunk.js                    276 kB   59.1 kB
dist/assets/js/chunk-vendors.ff8c2b62.chunk.js          2.01 MB  548 kB
dist/assets/js/chunk-vendors.ff8c2b62.chunk.js.LICENSE  74.1 kB  7.98 kB
dist/assets/js/runtime~app.4366c31c.js                  5.54 kB  2.29 kB
dist/default-favicon.ico                                5.43 kB  1.34 kB
dist/index.html                                         6.32 kB  2.68 kB
(total)                                                 6.13 MB  2.38 MB
```
### dist folder content

![image](https://user-images.githubusercontent.com/1560102/67074748-026d2900-f18a-11e9-8309-5f106eae7d83.png)

zcuric commented:
@vladimyr Build logs added. 

Is this viable now? 
> - Collection of former poi - now vue-cli plugins to get extracted to separate repo in order to decrease maintenance burden. That means keeping both poi versions until we migrate all our apps and this newly created vue-cli variants.

underscope commented:
@vladimyr

zcuric committed:
Move vue-cli plugins to specific folder

zcuric committed:
Merge branch 'develop' into tooling/poi-to-vue-cli

zcuric committed:
Move poi plugins to own folder

zcuric commented:
Conflicts resovled. 
### TODO
- [x] Update to latest `@vue/cli`
- [x] Update plugins to match lates version of plugins made for `poi`
- [ ] Check the build diff with develop
- [ ] QA the build

vladimyr commented on package.json:
```diff
@@ -181,5 +179,14 @@
+      "./build/plugins/vue-cli/clean-out-dir",
+      "./build/plugins/vue-cli/copy",
+      "./build/plugins/vue-cli/html-version-spec",
+      "./build/plugins/vue-cli/sass-resources"
```
Can we swap this with https://github.com/ExtensionEngine/vue-cli-plugin-sass-resources 🤔

zcuric replied:
Done

vladimyr commented:
CopyPlugin options need to be altered to match the new signature: https://app.circleci.com/pipelines/github/ExtensionEngine/tailor/1403/workflows/a383d39a-ddb3-4b20-9f1e-66ef3a520afd/jobs/1402/parallel-runs/0/steps/0-107

zcuric committed:
Merge branch 'develop' into tooling/poi-to-vue-cli

zcuric committed:
Add vue-cli-plugin-sass-resources

zcuric committed:
Update copy plugin to match new api

zcuric committed:
Update vue-cli, babel and core-js

zcuric committed:
Update ENV vars to be picked up by vue-cli

zcuric committed:
Add VUE_APP_TITLE to .env.example

zcuric committed:
Add VUE_APP_ENABLE_DEFAULT_SCHEMA

zcuric committed:
Add envs plugin :tada:

zcuric committed:
Add custom index.html with title

zcuric commented:
Build passing like a charm. 🥳  🎉

zcuric commented:
@underscope @vladimyr It's been a year and half since I've opened this PR. Can we close it at least so it doesn't pollute the PR tab, because I gave up any hope on this being merged? Maybe it should not be merge anyways, cause I don't see it how it will bring any meaningful value expect for maybe easier transition to Vue 3 and Vuetify 3. 🤷🏻

zcuric commented:
Closing this for now.
